jormungandr/src/explorer/persistent_sequence.rs

```diff
@@ -0,0 +1,36 @@
+use imhamt::Hamt;
+use std::collections::hash_map::DefaultHasher;
+
+// Use a Hamt to store a sequence, the indexes can be used for pagination
+// XXX: Maybe there is a better data structure for this?
+#[derive(Clone)]
+pub struct PersistentSequence<T> {
+    // Could be usize, but it's only used to store blocks and transactions for now
```

Comment:
Could there be more than 2^32 blocks at some future date, or is it within an epoch only?

In general, it only makes sense to be stingy on this if we seriously target wasm32 and small devices, otherwise just use `u64` wherever 1) it's not about a size of a data structure in memory; 2) you are not sure that the number is always going to be so ridiculously small as to not overflow `u32` even in intermediate calcualtions.